### PR TITLE
Add new `-histo` option to warp the newly-added PAM50 histology files

### LINKED ISSUE #2179: Add microstructure atlas to PAM50
Data are here: https://osf.io/8k7jr/
Paper: https://pubmed.ncbi.nlm.nih.gov/30326296/

Todo:
- bring the atlas in the same coordinate space as the PAM50
- maybe generate a subsampled version? (or in addition to?)
- update the course material
- update `sct_download_data`

### PR DESCRIPTION
**Pull request by alexfoias**
> **Note**: This PR was originally opened by @alexfoias. However, after @alexfoias left the lab, responsibility for this PR was transferred to @joshuacwnewton.

## Description

This PR adds a new CLI option to sct_warp_template (`-histo 1`) to warp the new histology files. 

This option works similarly to the existing options: `-a 1` (WM atlas) and `-s 1` (spinal levels). 

## Related work

Note: This PR depends on an updated release of the PAM50 template that includes the histology files (see https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3444#discussion_r1156123229).

## Linked issues

Fixes #2179.

### CODE REVIEW

**Comment by alexfoias** [comment]
@jcohenadad Do I also have to implement the test in CLI right ? https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/41126c604df500546f99056a6e66bd20d841dbd7/unit_testing/cli/test_cli_sct_warp_template.py

**Comment by joshuacwnewton** [comment]
> @jcohenadad Do I also have to implement the test in CLI right ? https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/41126c604df500546f99056a6e66bd20d841dbd7/unit_testing/cli/test_cli_sct_warp_template.py

That would be much appreciated. Currently, `sct_warp_template` is hardly tested at all (as you can see in that file) so almost any additions would be an improvement.

Note, though, that the existing test is currently being run on [`sct_testing_data/template`](https://github.com/spinalcordtoolbox/sct_testing_data/tree/master/template), which doesn't have all of the files from the full PAM50 template. So, we should maybe think about whether we want to update `sct_testing_data` with the files from https://github.com/spinalcordtoolbox/PAM50/pull/2, or run the tests directly on the full `$SCT_DIR/data/PAM50` template.

**Comment by joshuacwnewton** [comment]
It looks like this PR is running into https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3801. 

I have been able to test this PR locally, though, and the tests worked for me, so. I think this is close to being ready for review.

File: spinalcordtoolbox/scripts/sct_download_data.py [hunk 1]
```diff
@@ -42,8 +42,7 @@
     },
     "PAM50": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20201104/PAM50-r20201104.zip",
-            "https://osf.io/download/5fa21326a5bb9d00610a5a21/",
+            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20220527/PAM50-r20220527.zip",
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/scripts/sct_download_data.py:45]: The one downside to adding the histology microstructure atlas to the PAM50 template is that the installation size increases quite a bit:

* Before: [r20201104](https://github.com/spinalcordtoolbox/PAM50/releases/tag/r20201104) (67.6 MB)
* After: [r20220527](https://github.com/spinalcordtoolbox/PAM50/releases/tag/r20220527) (156 MB)

Is this an acceptable increase given the usefulness of the histology microstructure atlas?

Comment (jcohenadad) [reply spinalcordtoolbox/scripts/sct_download_data.py:45]: (from internal discussion): Might not need to be included in the mainline PAM50 template – could be a separate sct_download_data download.

File: spinalcordtoolbox/download.py [hunk 2]
```diff
@@ -39,8 +39,7 @@
     },
     "PAM50": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20201104/PAM50-r20201104.zip",
-            "https://osf.io/download/5fa21326a5bb9d00610a5a21/",
+            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20220527/PAM50-r20220527.zip",
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/download.py:42]: Note: Currently, this `r20220527` release is not up-to-date -- it contains the older, "uncorrected" histology files. So, we will need to create a new release that includes the corrected histology files.

However, we also need to coordinate this PAM50 release with the [Phillips Lab level files update](https://github.com/spinalcordtoolbox/PAM50/pull/3), too.  (Ideally, we would create a release that contains both updates.)

File: spinalcordtoolbox/download.py [hunk 3]
```diff
@@ -39,8 +39,7 @@
     },
     "PAM50": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20201104/PAM50-r20201104.zip",
-            "https://osf.io/download/5fa21326a5bb9d00610a5a21/",
+            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20220527/PAM50-histology-r20220527.zip",
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/download.py:42]: Note: Currently, this `r20220527` PAM50 release is not up-to-date -- it contains the older, "uncorrected" histology files. So, we will need to create a new release that includes the corrected histology files.

However, we also need to coordinate this PAM50 release with the [Phillips Lab level files update](https://github.com/spinalcordtoolbox/PAM50/pull/3), too.  (Ideally, we would create a release that contains both updates.)

File: spinalcordtoolbox/download.py [hunk 4]
```diff
@@ -39,8 +39,7 @@
     },
     "PAM50": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/PAM50/releases/download/r20201104/PAM50-r20201104.zip",
-            "https://osf.io/download/5fa21326a5bb9d00610a5a21/",
+            "https://github.com/spinalcordtoolbox/PAM50/archive/refs/heads/master.zip",
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/download.py:42]: Note: I've changed the PAM50 link to use the files in the `master` branch, that way we can quickly test @valosekj's excellent contributions in https://github.com/spinalcordtoolbox/PAM50/pull/6 and https://github.com/spinalcordtoolbox/PAM50/pull/9. 

But, ultimately, this link should be replaced by a proper release of the PAM50 template. (Ideally, one that includes both the histology files _and_ the updated Phillips level files.)

Comment (valosekj) [reply spinalcordtoolbox/download.py:42]: Thanks! 

Tested using the following commands and it seems to work well!

```console
# download PAM50 with histology atlas
git fetch
git checkout af/warp_histology_atlas
sct_download_data -d PAM50
# download testing data
sct_download_data -d sct_testing_data
# register t2 scan to PAM50
cd $SCT_DIR/data/sct_testing_data/t2
sct_register_to_template -i t2.nii.gz -s t2_seg-manual.nii.gz -lspinal t2_labeled_center_manual.nii.gz -c t2
# warp histology atlas
sct_warp_template -d t2.nii.gz -w warp_template2anat.nii.gz -histo 1 -a 0 -s 0
# check results
fsleyes t2.nii.gz label/histology/PAM50_200um_AVF.nii.gz
```

Comment (joshuacwnewton) [reply spinalcordtoolbox/download.py]: This has finally been resolved -- a new release of the PAM50 template (including the histology files) has been created: https://github.com/spinalcordtoolbox/PAM50/releases/tag/r20230519

Unless there are any other concerns, I think this PR can be merged?

**Pull request merged**